synapse/cores/common.py

```diff
@@ -1878,6 +1881,21 @@ def _runAutoAdd(self, toadd):
                 continue
             self.formTufoByProp(form, valu)
 
+    def _reqProps(self, form, fulls):
+        if not self.enforce:
+            return
+
+        props = self.getFormReqs(form)
+        if props and fulls.get('syn:prop:glob'):
+            # Special case for handling syn:prop:glob=1 on will not have a ptype
+            # despite the model requiring a ptype to be present.
+            props.pop('syn:prop:ptype', None)
+
+        for prop in props:
```

Comment:
to avoid the loops in python ( and make them be in C ) you could probably do something like
```
propset = set(props)
fullset = set(fulls)

missing = propset - fullset
if missing:
    etc etc etc
```

Reply:
This is fixed; along with a early return if there are no req'd props on the form.